src/arrayMethodSort.js

```diff
@@ -4,8 +4,42 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const length = this.length;
+
+    let comparator;
+
+    if (typeof compareFunction !== 'function') {
+      comparator = function (a, b) {
+        const strA = String(a);
```

Comment:
You can save some computational resources by converting the elements to string only once and storing them in variables.